snuba/query/parser/__init__.py

```diff
@@ -91,7 +96,10 @@ def build_selected_expressions(
 
     aggregations = []
     for aggregation in body.get("aggregations", []):
-        assert isinstance(aggregation, (list, tuple))
+        if not isinstance(aggregation, (list, tuple)):
+            raise ParsingException(
+                f"Invalid aggregation structure {aggregation}. It must be a list/tuple."
```

Comment:
If this error message is going to be propagated to the user, should this refer to a more abstract data type like a "sequence" instead of the specific Python types? (For that matter, should this `isinstance` check itself actually be on `collections.abc.Sequence` to check the interface and not implementation?)

Reply:
To be fair I did not make up my mind yet on whether these errors should directly be visible to the users or whether we should introduce an intermediate error code and mapping them to something user facing. But you are right, as long as it is propagated it should be python agnostic. Will change.

Reply:
Makes sense, I might have been reading between the lines a bit too much here.